src/main/java/domain/Lotto.java

```diff
@@ -0,0 +1,46 @@
+package domain;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Lotto {
+	public static final int LOTTO_NUMBER_SIZE = 6;
+
+	private final List<LottoNumber> lottoNumbers;
+
+	public Lotto(List<LottoNumber> lottoNumbers) {
+		validateSize(lottoNumbers);
+		validateDistinction(lottoNumbers);
+
+		Collections.sort(lottoNumbers);
+		this.lottoNumbers = lottoNumbers;
+	}
+
+	public boolean contains(LottoNumber lottoNumber) {
+		return lottoNumbers.contains(lottoNumber);
+	}
+
+	public int countSameNumber(Lotto lotto) {
+		return (int) lotto.getLottoNumbers().stream().filter(this::contains).count();
```

Comment:
한 줄에 하나의 함수만 호출하도록 아래와 같이 변경하면 조금 더 가독성이 좋아질 것 같네요. 🙂

```suggestion
		return (int) lotto.getLottoNumbers().stream()
		    .filter(this::contains)
		    .count();
```